ksql-parser/src/main/java/io/confluent/ksql/parser/rewrite/StatementRewriter.java

```diff
@@ -0,0 +1,1327 @@
+    return visitStatement(node, context);
+  }
+
+  protected Node visitShowColumns(final ShowColumns node, final Object context) {
+    return visitStatement(node, context);
+  }
+
+  protected Node visitShowPartitions(final ShowPartitions node, final Object context) {
+    return visitStatement(node, context);
+  }
+
+  protected Node visitShowCreate(final ShowCreate node, final Object context) {
+    return visitStatement(node, context);
+  }
+
+  protected Node visitShowFunctions(final ShowFunctions node, final Object context) {
+    return visitStatement(node, context);
+  }
+
+  protected Node visitSetSession(final SetSession node, final Object context) {
+    return visitStatement(node, context);
+  }
+
+  protected Node visitGenericLiteral(final GenericLiteral node, final Object context) {
+    return visitLiteral(node, context);
+  }
+
+  protected Node visitTimeLiteral(final TimeLiteral node, final Object context) {
+    return node.getLocation()
+        .map(location ->
+            new TimeLiteral(node.getLocation().get(), node.getValue())
+        )
+        .orElse(
+            new TimeLiteral(node.getValue())
+        );
+
+  }
+
+  protected Node visitExplainOption(final ExplainOption node, final Object context) {
+    return visitNode(node, context);
+  }
+
+  protected Node visitWithQuery(final WithQuery node, final Object context) {
+    return node.getLocation()
+        .map(location ->
+            new WithQuery(node.getLocation().get(),
+                node.getName(),
+                (Query) process(node.getQuery(), context),
+                node.getColumnNames())
+        )
+        .orElse(
+            new WithQuery(node.getName(),
+                (Query) process(node.getQuery(), context),
+                node.getColumnNames())
+        );
+
+  }
+
+  protected Node visitSelect(final Select node, final Object context) {
+    return node.getLocation()
+        .map(location ->
+            new Select(node.getLocation().get(),
+                node.isDistinct(),
+                node.getSelectItems()
+                    .stream()
+                    .map(selectItem -> (SelectItem) process(selectItem, context))
+                    .collect(Collectors.toList()))
+        )
+        .orElse(
+            new Select(node.isDistinct(),
+                node.getSelectItems()
+                    .stream()
+                    .map(selectItem -> (SelectItem) process(selectItem, context))
+                    .collect(Collectors.toList())
+            )
+        );
+  }
+
+  protected Node visitRelation(final Relation node, final Object context) {
+    return visitNode(node, context);
+  }
+
+  protected Node visitQueryBody(final QueryBody node, final Object context) {
+    return visitRelation(node, context);
+  }
+
+  protected Node visitQuerySpecification(final QuerySpecification node, final Object context) {
+    Optional<WindowExpression> windowExpression = node.getWindowExpression().isPresent()
```

Comment:
This can be simplified to:

```
Optional<WindowExpression> windowExpression = node.getWindowExpression()
   .flatMap((WindowExpression) process(node.getWindowExpression().get(), context));
```

Others elsewhere the same.

Reply:
We won't be able to use this since ` process(node.getWindowExpression().get(), context` does not return optional and we need to check `node.getWindowExpression()` before calling `get()`.

Reply:
Sorry, bad code example:

```
Optional<WindowExpression> windowExpression = node.getWindowExpression()
   .map(expression -> (WindowExpression) process(expression, context));
```